src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/AndInstruction.Boolean.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Linq.Expressions.Interpreter
+{
+    internal partial class AndInstruction
+    {
+        private sealed class AndBoolean : AndInstruction
```

Comment:
Split from the generated `And` instructions because this one has a custom implementation. Can push a separate PR (after https://github.com/dotnet/corefx/pull/14180 goes through) to reduce the noise here.